Assets/Scripts/Dungeon/Room Spawn.cs

```diff
@@ -0,0 +1,87 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class RoomSpawn : MonoBehaviour
+{
+    //private RoomTemplates template;
+    private RoomTemplates roomTemplates;
+    private checkroomcollision checkroom;
+    public GameObject deadEnd;
+    public GameObject template;
+
+    public bool collide;
+    public int attempts;
+
+    //public List<GameObject> Rooms;
```

Comment:
commented out code that you dont use for the final product should be removed before pull requests. It's good just to look through for them incase you realise later that you actually did need that still.